CMakeLists.txt

```
@@ -7,7 +7,11 @@ if(DEFINED CMAKE_JS_VERSION)
         elseif(CMAKE_GENERATOR_PLATFORM MATCHES "^[Xx]64$")
             set(VCPKG_TARGET_TRIPLET "x64-windows-static" CACHE STRING "")
         endif()
-        set(CMAKE_TOOLCHAIN_FILE "C:\\src\\vcpkg\\scripts\\buildsystems\\vcpkg.cmake")
+        if(DEFINED ENV{VCPKG_ROOT})
+            set(CMAKE_TOOLCHAIN_FILE "$ENV{VCPKG_ROOT}/scripts/buildsystems/vcpkg.cmake" CACHE STRING "")
```

```suggestion
            set(CMAKE_TOOLCHAIN_FILE "$ENV{VCPKG_ROOT}/scripts/buildsystems/vcpkg.cmake")
```
We don't need to save the variable in the cache.

---

I'm simply following the guide 🙂 Any reason this would hurt us?

---

It doesn't hurt, but it doesn't do anything much either. Normally you would use `set(CACHE)` if you don't want to overwrite a variable if it's already defined in a super scope. We don't have a super scope for this variable, though, and indeed setting it here should trump any other definitions. Technically, I needn't have set `VCPKG_TARGET_TRIPLET` like that either, and that's likely a bad copy/paste job on my part. Mind you, these variable are only considered by the `project()` command so beyond that their value is not considered.
If you do want to save it in the cache you should use `FILEPATH` instead of `STRING` as the entry type.